htdocs/core/modules/import/import_xlsx.modules.php

```diff
@@ -916,6 +916,10 @@ public function import_insert($arrayrecord, $array_match_file_to_database, $obji
 										$filters[] = $col.' = '.$data[$key];
 									}
 								}
+								if (!empty($tablewithentity_cache[$tablename])) {
+									$where[] = "entity = ".((int) $conf->entity);
```

Comment:
Ok, i see.
So I suggest to replace this line an dline 921 with
`$where[] = "entity IN (".getEntity($tablename).")";`
So when on session with entity 2 and 3 shared, if we find product PRODMULTI on 2 when on 3, update will work.
And if there is 2 products, one in 2 and same with same ref in 3, we will found 2 record and it will return the expected error (line 933) saying we don't know which record to update.

Reply:
@eldy 
Are you sure that it's always getEntity($tablename) with $tablename as parameter ?
In getEntity() method the first parameter is the object element and not table element.
And what about lines 961 and 1010 ? Should I also use "IN getEntity($tablename)" instead of "=" ?

Reply:
@eldy @lvessiller-opendsi 
this is not the "tablename" but this is the "element" name

Reply:
@eldy @hregis 
But how can I get the element name from table name ? (element is not defined in the declaration of import)
The getEntity with $tablename can solve the most of cases.

Reply:
@hregis 
But how can I get the element name from table name ?

Reply:
@lvessiller-opendsi @eldy 
the value of $tablename would be for example "llx_product"? 
and "getEntity()" does not handle table names.

So either we add the name of the element in the description of the import, eg:

```
$this->import_tables_array[$r] = array('p'=>MAIN_DB_PREFIX.'product');
$this->import_element_array[$r] = array('p'=>'product');

$this->import_tables_array[$r] = array('s'=>MAIN_DB_PREFIX.'societe');
$this->import_element_array[$r] = array('s'=>'thirdparty');
```

or I add a correspondence in the Multicompany module?

`array('llx_product' => 'product', 'llx_societe' => 'thirdparty');`

Reply:
@hregis @eldy 
I added a list to map table to element in model import class.
So I use this map list to filter with getEntity() method in import.
Could you check if it's a good way to implement it ?
I think I put all shareable tables for multi-company : could you check map list ? (hregis)

Reply:
@lvessiller-opendsi 
it's possible to call you by phone ? send me an email please, thanks

Reply:
@hregis 
Yes, I sent you an email with my phone number

Reply:
@eldy it's ok for me, thanks